packages/block-editor/src/components/use-setting/index.js

```diff
@@ -93,126 +94,158 @@ const removeCustomPrefixes = ( path ) => {
 	return prefixedFlags[ path ] || path;
 };
 
+const mergeCache = new WeakMap();
+function mergeOrigins( value ) {
+	let result = mergeCache.get( value );
+	if ( ! result ) {
+		result = [ 'default', 'theme', 'custom' ].flatMap(
+			( key ) => value[ key ] ?? []
+		);
+		mergeCache.set( value, result );
+	}
+	return result;
+}
+
 /**
  * Hook that retrieves the given setting for the block instance in use.
  *
  * It looks up the settings first in the block instance hierarchy.
  * If none is found, it'll look it up in the block editor store.
  *
- * @param {string} path The path to the setting.
- * @return {any} Returns the value defined for the setting.
- * @example
- * ```js
- * const isEnabled = useSetting( 'typography.dropCap' );
- * ```
+ * @param {string[]} paths The path to the setting.
+ * @return {any[]} Returns the values defined for the settings.
  */
-export default function useSetting( path ) {
-	const { name: blockName, clientId } = useBlockEditContext();
+export function useSettings( paths ) {
```

Comment:
Does it make sense to move `useSettings()` to a separate file? Perhaps under `../use-settings/index.js`? Right now it's a bit unfortunate that `use-setting/index.js` exports `useSettings()`.

Reply:
The hooks are very closely related, it's not justified to have them in separate files.

Most consumers should import them from the parent `components` module, not using the `/use-setting` subpath.

Reply:
I disagree because I still find it confusing and unexpected, but I'll agree to disagree and commit.